<?xml version="1.0"?>

<launch>

	<rosparam command="load" file="$(find crazyflie_human)/config/pedestrian_pred.yaml" />

	<!-- denotes which experimental setup we are running:  
		- coffee: one goal
		- triangle: two goals				
	-->
	<arg name="exp" default="coffee" />

	<!-- denotes which human model we are using:
		- irrational, rational, adaptive	
	-->
	<arg name="beta" default="adaptive" />

	<param name="exp" value="$(arg exp)" />
	<param name="beta" value="$(arg beta)" />

	<!-- simulate human walking around 
	<node name="simulate_human" pkg="crazyflie_human" type="sim_human.py" output="screen"/> -->
	
	<!-- read data from optitrack of human walking around -->
	<node name="human_state_estimator1" pkg="crazyflie_human" type="human_state_estimator.py" args="1" output="screen"/>
	<node name="human_state_estimator2" pkg="crazyflie_human" type="human_state_estimator.py" args="2" output="screen"/>

	<!-- predict each human -->
	<node name="human_prediction1" pkg="crazyflie_human" type="human_pred.py" args="1" output="screen"/> 
	<node name="human_prediction2" pkg="crazyflie_human" type="human_pred.py" args="2" output="screen"/> 
	
	<!-- predict all humans -->
	<node name="multi_human_prediction" pkg="crazyflie_human" type="multi_human_pred.py" output="screen"/> 

	<!-- visualize occupancy grids -->
	<node name="prediction_visualizer" pkg="crazyflie_human" type="prediction_visualizer.py" output="screen"/> 

	<!-- play rosbag of human data -->
	<!-- <arg name="path" default="$(find crazyflie_human)/data" />
	<arg name="file" default="sylvia" />
	<node pkg="rosbag" type="play" name="player" output="screen" args="$(arg path)$(arg exp)/$(arg file).bag"/> -->

</launch>
